.github/workflows/update_documentation.yml

```diff
@@ -45,6 +45,10 @@ jobs:
 # Setup and bootstrap
 #
     - uses: actions/checkout@v4
+      with: 
+        fetch-depth: 0
```

Comment:
What does this do? Was it actually necessary for the fix?

Reply:
Set fetch-depth: 0 to fetch all history for all branches and tags

I would think that we only need the history for the current branch. Actually I am not sure if we need them in any of our workflows